workspaces/adventure-pack/goodies/typescript/Iterator.prototype.reduce/index.ts

```diff
@@ -0,0 +1,38 @@
+import "../Iterator.prototype.toIterable";
+import { iteratorPrototype } from "../Iterator.prototype";
+
+declare global {
+  interface Iterator<T> {
+    reduce<U>(
+      this: Iterator<T>,
+      callbackFn: (accumulator: U, value: T, index: number) => U,
+      initialValue?: U,
+    ): U;
+  }
+}
+
+iteratorPrototype.reduce ??= function <T extends U, U>(
```

Comment:
I don't think it's necessary for T to extend U, is it? For example when we reduce an iterator of numbers to an object, `number` doesn't extend `Record`.

```suggestion
iteratorPrototype.reduce ??= function <T, U>(
```

Reply:
when I get rid of T extend U

I get this error
```
Type 'T' is not assignable to type 'U'.
  'U' could be instantiated with an arbitrary type which could be unrelated to 'T'.ts(2322)
index.ts(14, 40): This type parameter might need an `extends U` constraint.
```

Reply:
Ok, we can revisit this when we add the method signatures 🙂